src/e2c/pdf_tools.py

```diff
@@ -0,0 +1,165 @@
+"""Functions to handle highlights in PDF files.
+
+`e2c colors` should be used first to find all colors of highlights in
+the given PDF file. Then `e2c extract` can select all words of those
+highlights with the specified color.
+"""
+import pathlib
+import re
+import string
+
+import fitz
+import numpy as np
+from textblob import Word
+
+_threshold_color = 0.000001  # if two colors are different.
+_threshold_spell = 0.99  # if the word is spelled correctly.
+_threshold_intersection = 0.9  # if the intersection is large enough.
+_specials = "“" + "”"  # Some special punctuations to be removed.
+
+
+def _remove_punctuation(s):
+    """Remove all kinds of punctuations in the string.
+
+    Args:
+        s (str): to be removed from punctuations.
+
+    Returns:
+        str: with all kinds of punctuation removed.
+    """
+    table = str.maketrans({a: None for a in string.punctuation + _specials})
+    return s.translate(table)
+
+
+def _check_contain(r_word, points):
+    """If `r_word` is contained in the rectangular area.
+
+    The area of the intersection should be large enough compared to the
+    area of the given word.
+
+    Args:
+        r_word (fitz.Rect): rectangular area of a single word.
+        points (list): list of points in the rectangular area of the
+            given part of a highlight.
+
+    Returns:
+        bool: whether `r_word` is contained in the rectangular area.
+    """
+    # `r` is mutable, so everytime a new `r` should be initiated.
```

Comment:
remove this.

Reply:
There used to be an important bug. I wrote there in case I forgot in the future. Should I move to docstring instead?